### PR TITLE
Add node classification

### PR DESCRIPTION
**Pull request by jknipper**
This PR adds distinct names, properties and tags to nodes created by Kubernikus. Propertiers/metadata and tags will be reconciled on a timely basis.

**Update:** @databus23 refactored this PR quite a bit to improve performance and unify the tags/metadata.

New servers get created with the following settings:
* instance name is prefixed with `kks-`
* tags (single value):
  * `kubernikus`
  * `kubernikus:kluster=$NAME`
  * `kubernikus:nodepool=$NODEPOOL`
  * `kubernikus:template-version=$TEMPLATE_VERSION`
  * `kubernikus:api-version=$KLUSER_VERSION`
* metadata (key/value):
  * `provisioner` = `kubernikus`
  * `kubernikus:kluster` = `$NAME`
  * `kubernikus:nodepool` = `$NODEPOOL`
  *  `kubernikus:template-version` = `$TEMPLATE_VERSION`
  *  `kubernikus:api-version` = `$KUSTER_VERSION`
  
The flightcontroller reconciles the `kubernikus`, `kubernikus:kluster` and `kubernikus:nodepool` tags and metadata to instances so existing instances get retrofitted with those ones.
  This requires no additional requests to nova, as the already existing  instance list from the flightcontroller is reused for checking tags/metadata.

### CODE REVIEW

**Comment by BugRoger** [comment]
implements #569

**Comment by databus23** [comment]
rebased on master

**Comment by databus23** [comment]
refactored this PR substantially. @BugRoger Care to give it another review?

**Review by databus23 — APPROVED** [review]
I approve my refactor. Looks very good. :)

File: pkg/client/openstack/kluster/client.go [hunk 1]
```diff
@@ -73,24 +78,39 @@ func (c *klusterClient) CreateNode(kluster *v1.Kluster, pool *models.NodePool, n
...
+	}
+	imageID, err := imageutil.IDFromName(c.ImageClient, pool.Image)
+	if err != nil {
+		return "", fmt.Errorf("Failed to find id for image %s: %w", pool.Image, err)
+	}
+
+	tags := nodeTags(kluster.Spec.Name, pool.Name)
+	tags = append(tags, "kubernikus:template-version="+templates.TEMPLATE_VERSION)
+	tags = append(tags, "kubernikus:api-version="+kluster.Spec.Version)
+	metadata := nodeMetadata(kluster.Spec.Name, pool.Name)
```
Comment (BugRoger) [inline pkg/client/openstack/kluster/client.go:93]: Is `kluster.Name` == `kluster.Spec.Name`?

**Review by BugRoger — APPROVED** [review]
Nice. LGTM

File: pkg/client/openstack/kluster/client.go (hunk 1 above)
Comment (databus23) [reply pkg/client/openstack/kluster/client.go:93]: No, that's the unqualified name (without the project id), so the tag will be `kubernikus:kluster=s-qa-de-1` not `kubernikus:kluster=s-qa-de-1-10397130u4u10140109470921471024`. I figured its just noise as the server is running in the project anyways.

Comment (BugRoger) [reply pkg/client/openstack/kluster/client.go:93]: Cool. Open the flood-gates!

**Pull request merged**